src/coreclr/gc/gc.cpp

```diff
@@ -14114,6 +14220,46 @@ void allocator::thread_sip_fl (heap_segment* region)
 }
 #endif //USE_REGIONS
 
+#ifdef FEATURE_EVENT_TRACE
+uint16_t allocator::count_largest_items (uint32_t* bucket_info, 
+                                         size_t max_size, 
+                                         size_t max_item_count,
+                                         size_t* recorded_fl_info_size)
+{
+    assert (gen_number == max_generation);
+
+    size_t size_counted_total = 0;
+    size_t items_counted_total = 0;
+    for (int i = (num_buckets - 1); i >= 0; i--)
+    {
+        uint32_t items_counted = 0;
+        uint8_t* free_item = alloc_list_head_of ((unsigned int)i);
+        while (free_item)
+        {
+            assert (((CObjectHeader*)free_item)->IsFree());
+
+            size_t free_item_size = Align (size (free_item));
+            size_counted_total += free_item_size;
+            items_counted_total++;
+            items_counted++;
+
+            if ((size_counted_total > max_size) || (items_counted > max_item_count))
+            {
+                bucket_info[i] = items_counted;
```

Comment:
So you record the *number* of items per bucket - would it be more useful to record the *cumulative size* instead? Of course then uint32_t for the bucket_info may no longer be large enough...

Reply:
I could record the cumulative size per bucket in addition to the # of items in that bucket? the # is useful too, especially for larger buckets...

Reply:
Looks good, thanks!